cloud/scope/vpc_cluster.go

```diff
@@ -686,3 +742,326 @@ func (s *VPCClusterScope) buildCOSObjectHRef() (*string, error) {
+		subnets = s.IBMVPCCluster.Spec.Network.WorkerSubnets
+	}
+
+	// Reconcile Worker subnets.
+	for _, subnet := range subnets {
+		if requiresRequeue, err := s.reconcileSubnet(subnet, false); err != nil {
+			return false, fmt.Errorf("error failed reconciling worker subnet: %w", err)
+		} else if requiresRequeue {
+			// If the reconcile of the subnet requires further reconciliation, plan to requeue entire ReconcileSubnets call, but attempt to further reconcile additional Subnets (attempt all subnet reconciliation).
+			requeue = true
+		}
+	}
+
+	// Return whether or not one or more subnets required further reconciling after attempting to process all Control Plane and Worker subnets.
+	return requeue, nil
+}
+
+// reconcileSubnet will attempt to find the existing subnet, or create it if necessary.
+// The logic can handle either Control Plane or Worker subnets, but must distinguish between them for Status updates.
+func (s *VPCClusterScope) reconcileSubnet(subnet infrav1beta2.Subnet, isControlPlane bool) (bool, error) { //nolint: gocyclo
+	// Check Status first and update as necessary.
+	if s.NetworkStatus() != nil {
+		var subnetMap map[string]*infrav1beta2.ResourceStatus
+		var subnetID, subnetName *string
+		if isControlPlane && s.NetworkStatus().ControlPlaneSubnets != nil {
+			subnetMap = s.NetworkStatus().ControlPlaneSubnets
+		} else if !isControlPlane && s.NetworkStatus().WorkerSubnets != nil {
+			subnetMap = s.NetworkStatus().WorkerSubnets
+		}
+		// Based on Network Status, setup either the name or ID for lookup of the subnet's current status.
+		if subnet.Name != nil {
+			if _, ok := subnetMap[*subnet.Name]; ok {
+				subnetName = subnet.Name
+			}
+		} else if subnetID != nil && subnet.ID != nil {
+			for _, statusSubnet := range subnetMap {
+				if statusSubnet.ID == *subnet.ID {
+					subnetID = subnet.ID
+				}
+			}
+		}
+
+		// Perform current status lookup of subnet, using ID or name if one was found in Network Status.
+		if subnetID != nil {
+			options := &vpcv1.GetSubnetOptions{
+				ID: subnetID,
+			}
+			subnetDetails, _, err := s.VPCClient.GetSubnet(options)
+			if err != nil {
+				return false, fmt.Errorf("error retrieving existing subnet by id %s: %w", *subnetID, err)
+			} else if subnetDetails == nil {
+				return false, fmt.Errorf("error failed to find existing subnet by id %s", *subnetID)
+			}
+			return s.updateSubnetStatus(subnetDetails, isControlPlane)
+		} else if subnetName != nil {
+			subnetDetails, err := s.VPCClient.GetVPCSubnetByName(*subnetName)
+			if err != nil {
+				return false, fmt.Errorf("error retrieving existing subnet by name %s: %w", *subnetName, err)
+			} else if subnetDetails == nil {
+				return false, fmt.Errorf("error failed to find existing subnet by name: %s", *subnetName)
+			}
+			return s.updateSubnetStatus(subnetDetails, isControlPlane)
+		}
+	}
+
+	// Otherwise, if these is an ID or name, attempt to lookup the subnet and update status as necessary.
+	if subnet.ID != nil {
```

Comment:
If you found an existing subnet with id or name, wondering how are you planning to differentiate this during delete phase? Just asking for my understanding!
If it via tags, hope all the resources involved are having CRN like you mentioned, to attach the tags.

Reply:
At this moment, I don't have solid plans for deletion logic. Tagging is going to likely be involved/required to properly handle all of the edge cases involved with bring your own 'resource' (vpc, subnet, image, etc.).
Yes, I have already gone through and tested the tagging required for the resource CAPI is creating, but deletion has not been a focus, given time constraints.